client/src/main/java/org/spine3/base/ListStringifier.java

```diff
@@ -0,0 +1,201 @@
+        elementDelimiterPattern = createElementDelimiterPattern(delimiter);
+    }
+
+    private static String createElementDelimiterPattern(char delimiter) {
+        return Pattern.compile("(?<!\\\\)\\\\\\" + delimiter)
+                      .pattern();
+    }
+
+    @Override
+    protected String toString(List<T> list) {
+        final Function<T, QuotedListItem<T>> function = new Function<T, QuotedListItem<T>>() {
+            @Nullable
+            @Override
+            public QuotedListItem<T> apply(@Nullable T input) {
+                if(input == null){
+                    throw newIllegalArgumentException("The list element cannot be null.");
+                }
+                return of(input);
+            }
+        };
+        final List<QuotedListItem<T>> quotedItems = Lists.transform(list, function);
+        final String result = Joiner.on(delimiter)
+                                    .join(quotedItems);
+        return result;
+    }
+
+    @Override
+    protected List<T> fromString(String s) {
+        final String escapedString = escaper.escape(s);
+        final Splitter splitter = Splitter.onPattern(elementDelimiterPattern);
+        final List<String> elements = newArrayList(splitter.split(escapedString));
+        final Function<String, T> function = new Function<String, T>() {
+            @SuppressWarnings("ConstantConditions")
+            // It should be processed by the stringifier.
+            @Nullable
+            @Override
+            public T apply(@Nullable String input) {
+                return parse(input, listGenericClass);
+            }
+        };
+        final List<T> result = newArrayList(Lists.transform(elements, function));
+        return result;
+    }
+
+    /**
+     * Encloses and discloses each element of the list into and from quotes
+     *
+     * @param <T> the type of the elements in the list
+     */
+    static class QuotedListItem<T> extends Stringifiers.QuotedItem {
+
+        private final T object;
+        private final Class<T> genericClass;
+
+        @SuppressWarnings("unchecked")
+        // It is OK because the class is same.
+        private QuotedListItem(T object) {
+            this.object = object;
+            this.genericClass = (Class<T>) object.getClass();
+        }
+
+        static <T> QuotedListItem<T> of(T object) {
+            return new QuotedListItem<>(object);
+        }
+
+        static <T> T parse(String elementToParse, Class<T> genericClass) {
+            checkElement(elementToParse);
+            return convert(unquote(elementToParse), genericClass);
+        }
+
+        private static void checkElement(CharSequence element) {
+            final boolean isQuoted = isQuotedString(element);
+            if (!isQuoted) {
+                throw newIllegalArgumentException("Illegal format of the element: " + element);
+            }
+        }
+
+        @Override
+        public String toString() {
+            if (isString(genericClass)) {
```

Comment:
It seems like this is a responsibility of `Stringifiers.toString(object, genericClass)`.

You could save yourselves some `if`s here and in the `MapStringifier` class as well.